pretext/ExtraTopics/Glossary.ptx

```diff
@@ -0,0 +1,156 @@
+                            all programming languages are formal languages.</p>
+    </gi>
+    <gi>
+      <title>high-level language</title>
+      <p>A programming language like Python that is designed to be easy for
+                            humans to read and write.</p>
+    </gi>
+    <gi>
+      <title>interpret</title>
+      <p>To execute a program in a high-level language by translating it one
+                            line at a time.</p>
+    </gi>
+    <gi>
+      <title>low-level language</title>
+      <p>A programming language that is designed to be easy for a computer to
+                            execute; also called machine language or assembly language.</p>
+    </gi>
+    <gi>
+      <title>natural language</title>
+      <p>Any one of the languages that people speak that evolved naturally.</p>
+    </gi>
+    <gi>
+      <title>object code</title>
+      <p>The output of the compiler after it translates the program.</p>
+    </gi>
+    <gi>
+      <title>parse</title>
+      <p>To examine a program and analyze the syntactic structure.</p>
+    </gi>
+    <gi>
+      <title>portability</title>
+      <p>A property of a program that can run on more than one kind of computer.</p>
+    </gi>
+    <gi>
+      <title>print function</title>
+      <p>A function used in a program or script that causes the Python
+                            interpreter to display a value on its output device.</p>
+    </gi>
+    <gi>
+      <title>problem solving</title>
+      <p>The process of formulating a problem, finding a solution, and
+                            expressing the solution.</p>
+    </gi>
+    <gi>
+      <title>program</title>
+      <p>A sequence of instructions that specifies to a computer actions and
+                            computations to be performed.</p>
+    </gi>
+    <gi>
+      <title>programming language</title>
+      <p>A formal notation for representing solutions.</p>
+    </gi>
+    <gi>
+      <title>Python shell</title>
+      <p>An interactive user interface to the Python interpreter. The user of a
+                            Python shell types commands at the prompt (&gt;&gt;&gt;), and presses the return
+                            key to send these commands immediately to the interpreter for
+                            processing.</p>
+    </gi>
+    <gi>
+      <title>runtime error</title>
+      <p>An error that does not occur until the program has started to execute
+                            but that prevents the program from continuing.</p>
+    </gi>
+    <gi>
+      <title>semantic error</title>
+      <p>An error in a program that makes it do something other than what the
+                            programmer intended.</p>
+    </gi>
+    <gi>
+      <title>semantics</title>
+      <p>The meaning of a program.</p>
+    </gi>
+    <gi>
+      <title>shell mode</title>
+      <p>A style of using Python where we type expressions at the command
```

Comment:
I really don't like this - I think this should be deleted and rolled into the Python Shell definition.